src/ray/core_worker/task_manager.cc

```diff
@@ -917,6 +945,11 @@ void TaskManager::RemoveFinishedTaskReferences(
       return_ids.push_back(dynamic_return_id);
     }
   }
+  if (spec.IsStreamingGenerator()) {
```

Comment:
This is necessary because when we know NumStreamingGeneratorReturns (meaning the task has been finished before), we want to update the finished task references for those references.

Reply:
Could you convert this to a comment